### ISSUE TITLE
FAIL Installing Net::Whois::RIPE failed (Net-Whois-RIPE-2.006002.tar.gz)

### ISSUE THREAD

**Issue by netravnen** [arhuman/Net-Whois-RIPE#23]
Trying to install [rpsltool](https://github.com/rfc1036/rpsltool). And using `Net::Whois::RIPE` as a dependency. But for some reason when running `sudo cpanm Net::Whois::RIPE` to install the module. I am hit with the below error.

As I am not a perl expert by any means. And not using perl based projects often. I unsure what to look for when diagnosing the error. Hence why I have crearted this issue on the bugtracker.

```
$ sudo cat /root/.cpanm/work/1536331547.23709/build.log
cpanm (App::cpanminus) 1.7044 on perl 5.024001 built for x86_64-linux-gnu-thread-multi
Work directory is /root/.cpanm/work/1536331547.23709
You have make /usr/bin/make
You have LWP 6.15
You have /bin/tar: tar (GNU tar) 1.29
Copyright (C) 2015 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>.
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

Written by John Gilmore and Jay Fenlason.
You have /usr/bin/unzip
Searching Net::Whois::RIPE () on cpanmetadb ...
--> Working on Net::Whois::RIPE
Fetching http://www.cpan.org/authors/id/A/AA/AASSAD/Net-Whois-RIPE-2.006002.tar.gz
-> OK
Unpacking Net-Whois-RIPE-2.006002.tar.gz
Entering Net-Whois-RIPE-2.006002
Checking configure dependencies from META.json
Checking if you have ExtUtils::MakeMaker 6.58 ... Yes (7.10_02)
Configuring Net-Whois-RIPE-2.006002
Running Makefile.PL
Checking if your kit is complete...
Looks good
Generating a Unix-style Makefile
Writing Makefile for Net::Whois::RIPE
Writing MYMETA.yml and MYMETA.json
-> OK
Checking dependencies from MYMETA.json ...
Checking if you have Iterator 0 ... Yes (0.03)
Checking if you have Test::More 0 ... Yes (1.001014)
Building and testing Net-Whois-RIPE-2.006002
cp lib/Net/Whois/Object/FilterSet/AFRINIC.pm blib/lib/Net/Whois/Object/FilterSet/AFRINIC.pm
cp lib/Net/Whois/Object/Irt.pm blib/lib/Net/Whois/Object/Irt.pm
cp lib/Net/Whois/Object/FilterSet.pm blib/lib/Net/Whois/Object/FilterSet.pm
cp lib/Net/Whois/Object/AsBlock/AFRINIC.pm blib/lib/Net/Whois/Object/AsBlock/AFRINIC.pm
cp lib/Net/Whois/Object/Domain/AFRINIC.pm blib/lib/Net/Whois/Object/Domain/AFRINIC.pm
cp lib/Net/Whois/Object/InetNum/AFRINIC.pm blib/lib/Net/Whois/Object/InetNum/AFRINIC.pm
cp lib/Net/Whois/Object/Information.pm blib/lib/Net/Whois/Object/Information.pm
cp lib/Net/Whois/Object/AsSet.pm blib/lib/Net/Whois/Object/AsSet.pm
cp lib/Net/Whois/Object/InetNum/APNIC.pm blib/lib/Net/Whois/Object/InetNum/APNIC.pm
cp lib/Net/Whois/Object/Inet6Num/APNIC.pm blib/lib/Net/Whois/Object/Inet6Num/APNIC.pm
cp lib/Net/Whois/Object/KeyCert.pm blib/lib/Net/Whois/Object/KeyCert.pm
cp lib/Net/Whois/Object/AsSet/APNIC.pm blib/lib/Net/Whois/Object/AsSet/APNIC.pm
cp lib/Net/Whois/Object.pm blib/lib/Net/Whois/Object.pm
cp lib/Net/Whois/Object/InetRtr/AFRINIC.pm blib/lib/Net/Whois/Object/InetRtr/AFRINIC.pm
cp lib/Net/Whois/Object/Domain.pm blib/lib/Net/Whois/Object/Domain.pm
cp lib/Net/Whois/Object/Domain/APNIC.pm blib/lib/Net/Whois/Object/Domain/APNIC.pm
cp lib/Net/Whois/Object/InetRtr.pm blib/lib/Net/Whois/Object/InetRtr.pm
cp lib/Net/Whois/Object/InetNum.pm blib/lib/Net/Whois/Object/InetNum.pm
cp lib/Net/Whois/Object/AsBlock.pm blib/lib/Net/Whois/Object/AsBlock.pm
cp lib/Net/Whois/Object/AutNum/APNIC.pm blib/lib/Net/Whois/Object/AutNum/APNIC.pm
cp lib/Net/Whois/Object/Inet6Num.pm blib/lib/Net/Whois/Object/Inet6Num.pm
cp lib/Net/Whois/Object/KeyCert/APNIC.pm blib/lib/Net/Whois/Object/KeyCert/APNIC.pm
cp lib/Net/Whois/Object/AsSet/AFRINIC.pm blib/lib/Net/Whois/Object/AsSet/AFRINIC.pm
cp lib/Net/Whois/Object/AutNum/AFRINIC.pm blib/lib/Net/Whois/Object/AutNum/AFRINIC.pm
cp lib/Net/Whois/Object/Limerick.pm blib/lib/Net/Whois/Object/Limerick.pm
cp lib/Net/Whois/Object/Inet6Num/AFRINIC.pm blib/lib/Net/Whois/Object/Inet6Num/AFRINIC.pm
cp lib/Net/Whois/Object/Irt/APNIC.pm blib/lib/Net/Whois/Object/Irt/APNIC.pm
cp lib/Net/Whois/Object/Irt/AFRINIC.pm blib/lib/Net/Whois/Object/Irt/AFRINIC.pm
cp lib/Net/Whois/Object/FilterSet/APNIC.pm blib/lib/Net/Whois/Object/FilterSet/APNIC.pm
cp lib/Net/Whois/Generic.pm blib/lib/Net/Whois/Generic.pm
cp lib/Net/Whois/Object/Information/AFRINIC.pm blib/lib/Net/Whois/Object/Information/AFRINIC.pm
cp lib/Net/Whois/Object/AsBlock/APNIC.pm blib/lib/Net/Whois/Object/AsBlock/APNIC.pm
cp lib/Net/Whois/Object/KeyCert/AFRINIC.pm blib/lib/Net/Whois/Object/KeyCert/AFRINIC.pm
cp lib/Net/Whois/Object/AutNum.pm blib/lib/Net/Whois/Object/AutNum.pm
cp lib/Net/Whois/Object/PeeringSet.pm blib/lib/Net/Whois/Object/PeeringSet.pm
cp lib/Net/Whois/Object/Person/APNIC.pm blib/lib/Net/Whois/Object/Person/APNIC.pm
cp lib/Net/Whois/Object/RouteSet.pm blib/lib/Net/Whois/Object/RouteSet.pm
cp lib/Net/Whois/Object/PoeticForm/APNIC.pm blib/lib/Net/Whois/Object/PoeticForm/APNIC.pm
cp lib/Net/Whois/Object/Route6.pm blib/lib/Net/Whois/Object/Route6.pm
cp lib/Net/Whois/Object/RtrSet.pm blib/lib/Net/Whois/Object/RtrSet.pm
cp lib/Net/Whois/Object/Route6/APNIC.pm blib/lib/Net/Whois/Object/Route6/APNIC.pm
cp lib/Net/Whois/Object/Limerick/AFRINIC.pm blib/lib/Net/Whois/Object/Limerick/AFRINIC.pm
cp lib/Net/Whois/Object/Poem.pm blib/lib/Net/Whois/Object/Poem.pm
cp lib/Net/Whois/Object/Person/AFRINIC.pm blib/lib/Net/Whois/Object/Person/AFRINIC.pm
cp lib/Net/Whois/Object/Mntner.pm blib/lib/Net/Whois/Object/Mntner.pm
cp lib/Net/Whois/Object/Route/APNIC.pm blib/lib/Net/Whois/Object/Route/APNIC.pm
cp lib/Net/Whois/Object/Role.pm blib/lib/Net/Whois/Object/Role.pm
cp lib/Net/Whois/Object/Poem/APNIC.pm blib/lib/Net/Whois/Object/Poem/APNIC.pm
cp lib/Net/Whois/Object/Organisation.pm blib/lib/Net/Whois/Object/Organisation.pm
cp lib/Net/Whois/Object/Role/AFRINIC.pm blib/lib/Net/Whois/Object/Role/AFRINIC.pm
cp lib/Net/Whois/Object/PeeringSet/AFRINIC.pm blib/lib/Net/Whois/Object/PeeringSet/AFRINIC.pm
cp lib/Net/Whois/Object/Mntner/AFRINIC.pm blib/lib/Net/Whois/Object/Mntner/AFRINIC.pm
cp lib/Net/Whois/Object/Route6/AFRINIC.pm blib/lib/Net/Whois/Object/Route6/AFRINIC.pm
cp lib/Net/Whois/Object/Route/AFRINIC.pm blib/lib/Net/Whois/Object/Route/AFRINIC.pm
cp lib/Net/Whois/Object/PoeticForm.pm blib/lib/Net/Whois/Object/PoeticForm.pm
cp lib/Net/Whois/Object/RouteSet/AFRINIC.pm blib/lib/Net/Whois/Object/RouteSet/AFRINIC.pm
cp lib/Net/Whois/Object/Organisation/APNIC.pm blib/lib/Net/Whois/Object/Organisation/APNIC.pm
cp lib/Net/Whois/Object/RtrSet/AFRINIC.pm blib/lib/Net/Whois/Object/RtrSet/AFRINIC.pm
cp lib/Net/Whois/Object/PeeringSet/APNIC.pm blib/lib/Net/Whois/Object/PeeringSet/APNIC.pm
cp lib/Net/Whois/Object/RouteSet/APNIC.pm blib/lib/Net/Whois/Object/RouteSet/APNIC.pm
cp lib/Net/Whois/Object/Response.pm blib/lib/Net/Whois/Object/Response.pm
cp lib/Net/Whois/Object/Role/APNIC.pm blib/lib/Net/Whois/Object/Role/APNIC.pm
cp lib/Net/Whois/Object/Mntner/APNIC.pm blib/lib/Net/Whois/Object/Mntner/APNIC.pm
cp lib/Net/Whois/Object/RtrSet/APNIC.pm blib/lib/Net/Whois/Object/RtrSet/APNIC.pm
cp lib/Net/Whois/Object/Person.pm blib/lib/Net/Whois/Object/Person.pm
cp lib/Net/Whois/Object/Route.pm blib/lib/Net/Whois/Object/Route.pm
cp lib/Net/Whois/Object/Organisation/AFRINIC.pm blib/lib/Net/Whois/Object/Organisation/AFRINIC.pm
cp lib/Net/Whois/RIPE.pm blib/lib/Net/Whois/RIPE.pm
Manifying 28 pod documents
Manifying 28 pod documents
Manifying 12 pod documents
PERL_DL_NONLAZY=1 PERL_USE_UNSAFE_INC=1 "/usr/bin/perl" "-MExtUtils::Command::MM" "-MTest::Harness" "-e" "undef *Test::Harness::Switches; test_harness(0, 'blib/lib', 'blib/arch')" t/*.t
t/00-changes.t ................ skipped: Test::CPAN::Changes required for this test
# Testing Net::Whois::RIPE 2.006002, Perl 5.024001, /usr/bin/perl
t/00-load.t ................... ok
t/01-instantiation.t .......... ok
t/02-methods.t ................ ok
t/03-objects.t ................ ok
t/04-generic.t ................ ok
t/105-AsBlock.t ............... ok
t/110-AutNum.t ................ ok
t/115-Person.t ................ ok
t/120-Role.t .................. ok
t/125-AsSet.t ................. ok
t/130-Domain.t ................ ok
t/135-InetNum.t ............... ok
t/140-Inet6Num.t .............. ok
t/145-InetRtr.t ............... ok
t/150-RtrSet.t ................ ok
t/155-Mntner.t ................ ok
t/160-KeyCert.t ............... ok
t/165-Route.t ................. ok
t/167-Route6.t ................ ok
t/170-RouteSet.t .............. ok
t/175-PeeringSet.t ............ ok
t/180-Limerick.t .............. ok
t/185-Poem.t .................. ok
t/187-PoeticForm.t ............ ok
t/190-Organisation.t .......... ok
t/195-Response.t .............. ok
t/200-Information.t ........... ok
t/205-Irt.t ................... ok
t/210-FilterSet.t ............. ok


Set TEST_MNTNER, TEST_MNTNER_PASSWORD environment vars for live testing
    TEST_MNTNER being a maintener's nic-hdl in the RIPE test database
    TEST_MNTNER_PASSWORD being its password

t/215-SyncUpdates.t ........... skipped: Set environment vars for server testing

Set TEST_MNTNER, TEST_MNTNER_PGPKEY environment vars for live testing
    TEST_MNTNER being a maintener's nic-hdl in the RIPE test database
    TEST_MNTNER_PGPKEY being the key ID of an key-cert associated with
    the maintainer object
t/216-SyncUpdates-Signed.t .... skipped: Set environment vars for server testing
t/300-AFRINIC.t ............... ok
Unknown method 'last_modified' for object Net::Whois::Object::Mntner::APNIC (Did the Database schema changed ?) at /root/.cpanm/work/1536331547.23709/Net-Whois-RIPE-2.006002/blib/lib/Net/Whois/Generic.pm line 552.

#   Failed test 'Client performs queries without dying Can't require Net::Whois::Object::Mntner::IDNIC (No such file or directory) at /root/.cpanm/work/1536331547.23709/Net-Whois-RIPE-2.006002/blib/lib/Net/Whois/Object.pm line 693, <GEN0> line 7.
# '
#   at t/305-APNIC.t line 17.

#   Failed test 'undef isa 'Net::Whois::Object::Mntner::APNIC''
#   at t/305-APNIC.t line 18.
#     undef isn't defined
Unknown method 'last_modified' for object Net::Whois::Object::InetNum::APNIC (Did the Database schema changed ?) at /root/.cpanm/work/1536331547.23709/Net-Whois-RIPE-2.006002/blib/lib/Net/Whois/Generic.pm line 552.
# Looks like you failed 2 tests of 10.
t/305-APNIC.t ................. 
Dubious, test returned 2 (wstat 512, 0x200)
Failed 2/10 subtests 
t/boilerplate.t ............... skipped: Author tests not required to build this module
t/bug-disconnect.t ............ ok
t/bug-hostname-constructor.t .. ok
t/pod-coverage.t .............. skipped: Author tests not required to build this module
t/pod.t ....................... skipped: Author tests not required to build this module

Test Summary Report
-------------------
t/305-APNIC.t               (Wstat: 512 Tests: 10 Failed: 2)
  Failed tests:  3-4
  Non-zero exit status: 2
Files=39, Tests=3679,  6 wallclock secs ( 0.54 usr  0.09 sys +  3.75 cusr  0.74 csys =  5.12 CPU)
Result: FAIL
Failed 1/39 test programs. 2/3679 subtests failed.
Makefile:1238: recipe for target 'test_dynamic' failed
make: *** [test_dynamic] Error 255
-> FAIL Installing Net::Whois::RIPE failed. See /root/.cpanm/work/1536331547.23709/build.log for details. Retry with --force to force install it.
```

**Comment by netravnen** [comment]
I see this has a connection with the already posted error in #22

**Comment by arhuman** [comment]
Fixed in 2.007000
